src/Interfaces.ts

```diff
@@ -1,72 +1,136 @@
-    readonly [key: string]: string
-  }
+export interface ClaimContext {
+  readonly [key: string]: unknown
 }
 
-export function isClaim(object: any): object is Claim {
-  // TODO: use joi or protobuf
-  return !!(object.id && object.publicKey && object.signature && object.type && object.attributes && object.created)
+export interface ClaimTypeContexts {
+  readonly [key: string]: ClaimContext
 }
 
 // WARNING: This MUST account for ALL of the attributes in a Claim, except for id, @context, and signature.
 // Otherwise those attributes without context will be left out of the canonized/signed claim.
 // Refer to https://www.w3.org/2018/jsonld-cg-reports/json-ld/#the-context
-export const ClaimContext: ClaimContext = {
-  '@context': {
-    attributes: 'http://schema.org/CreativeWork',
-    author: 'http://schema.org/author',
-    text: 'http://schema.org/text',
-    created: 'http://purl.org/dc/terms/created',
-    dateCreated: 'http://schema.org/dateCreated',
-    datePublished: 'http://schema.org/datePublished',
-    name: 'http://schema.org/name',
-    tags: 'http://schema.org/keywords',
-    type: 'http://schema.org/additionalType',
-    publicKey: 'http://schema.org/Text',
-  },
+export const DefaultClaimContext: ClaimContext = {
+  cred: 'https://w3id.org/credentials#',
+  dc: 'http://purl.org/dc/terms/',
+  schema: 'http://schema.org/',
+  sec: 'https://w3id.org/security#',
+
+  id: 'sec:digestValue',
+  issuer: 'cred:issuer',
+  issuanceDate: 'cred:issued',
+  type: 'schema:additionalType',
+  claim: 'schema:Thing', // The most generic definition in schema.org,
 }
 
-export interface ClaimAttributes {
-  readonly [key: string]: string
+export const DefaultWorkClaimContext: ClaimContext = {
+  archiveUrl: 'schema:url',
+  author: 'schema:author',
+  canonicalUrl: 'schema:url',
+  claim: 'schema:CreativeWork',
+  contributors: {
+    '@id': 'schema:ItemList',
+    '@container': '@list',
+    '@type': 'schema:contributor',
+  },
+  copyrightHolder: 'schema:copyrightHolder',
+  dateCreated: 'schema:dateCreated',
+  datePublished: 'schema:datePublished',
+  license: 'schema:license',
+  name: 'schema:name',
+  tags: 'schema:keywords',
+  hash: 'sec:digestValue',
 }
 
-export enum ClaimType {
-  Identity = 'Identity',
-  Work = 'Work',
+export const DefaultIdentityClaimContext: ClaimContext = {
+  publicKey: 'sec:publicKeyBase58',
+  profileUrl: 'sec:owner',
 }
 
-export interface WorkAttributes extends ClaimAttributes {
-  readonly name: string
-  readonly datePublished: string
-  readonly dateCreated: string
-  readonly author: string
-  readonly tags?: string
-  readonly text: string
+export const claimTypeDefaults: ClaimTypeContexts = {
+  Work: DefaultWorkClaimContext,
+  Identity: DefaultIdentityClaimContext,
 }
 
-export interface Work extends Claim<WorkAttributes> {}
+const signatureSchema = {
+  '@graph': Joi.object().keys({
+    '@type': Joi.string().only(Object.keys(jsigs.suites).map(suite => `sec:${suite}`)),
```

Comment:
If `signatureSchema` needs the `jsigs` instance, it should live inside the object that wraps it... but maybe this can refer to the singleton `JSIG` to avoid that? Is `jsigs.suites` affected by anything, or is it some sort of universal constant that never changes?

Reply:
A universal constant that doesn't change. However, I don't know how to export both a factory and constant with the `export =`